### PR TITLE
fix(Upgradable): improved contract id verification

### PR DESCRIPTION
**Pull request by re1ro**
* [x] `Upgradable`: checking contract id against current implementation when upgrading
* [x] `InterchainAddressTracker`: prettier

### CODE REVIEW

File: contracts/upgradable/Implementation.sol [hunk 1]
```diff
@@ -10,14 +10,10 @@ import { IImplementation } from '../interfaces/IImplementation.sol';
...
-    address private immutable implementationAddress;
-
     /**
-     * @dev Contract constructor that sets the implementation address to the address of this contract.
+     * @dev Sets the implementation address to the address of this contract.
      */
-    constructor() {
-        implementationAddress = address(this);
-    }
+    address private immutable implementationAddress = address(this);
```
Comment (deanamiel) [inline contracts/upgradable/Implementation.sol:16]: Can this not be declared as a `constant` variable?

**Review by deanamiel — APPROVED** [review]

File: contracts/upgradable/Implementation.sol (hunk 1 above)
Comment (re1ro) [reply contracts/upgradable/Implementation.sol:16]: Constant is defined at the compilation time and deployed address would be unknown, immutable is defined at the deployment time

File: contracts/upgradable/Implementation.sol [hunk 2]
```diff
@@ -10,14 +10,10 @@ import { IImplementation } from '../interfaces/IImplementation.sol';
  * @dev Derived contracts must implement the setup function.
  */
 abstract contract Implementation is IImplementation {
-    address private immutable implementationAddress;
-
     /**
-     * @dev Contract constructor that sets the implementation address to the address of this contract.
+     * @dev Sets the implementation address to the address of this contract.
```
Comment (milapsheth) [inline contracts/upgradable/Implementation.sol:14]: we decided to revert this file change

File: contracts/upgradable/Implementation.sol [hunk 3]
```diff
@@ -10,14 +10,10 @@ import { IImplementation } from '../interfaces/IImplementation.sol';
  * @dev Derived contracts must implement the setup function.
```
Comment (milapsheth) [inline contracts/upgradable/Implementation.sol:10]: bump version in package.json

**Review by milapsheth — APPROVED** [review]

**Pull request merged**